### PR TITLE
Feat: Auto-Tests enabled

### PR DESCRIPTION
**Pull request by TfedUD**
Enabled 
     ```on: [push, pull_request]```
 automatic tests

*** also added dependabot config: 
     [refer to guide to confirm setup enabled in main repo](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/)

I did this coz I had to manually update dependencies to get the tests to pass: and figured this would keep you and whoever from needing to do manually ever again lol

### CODE REVIEW

**Comment by TfedUD** [comment]
@PH-Tools OK so tests action and coverage action are a go.
when merged; ```on: [push, pull_request]``` they both will run and via the actions tab of the repository you can see in detail the terminal processes and reporting. 

in ```tests\.coveragerc``` I had to specify the fail under float() number otherwise it would fail @ 98% coz was not 100% lol.
its currently @ 80.00% ready for you to specify what you want explicitly.
here is the coverage .coveragerc (config file) [documentation link](https://coverage.readthedocs.io/en/latest/config.html#report-fail-under)

# TLDR:
has coverage action
has unit test action
has dependabot (may need some additional config through repo management interface)

**Review by PH-Tools — APPROVED** [review]
Lets give it a whirl!!
